### ISSUE TITLE
Implement shrinking in Map resize

### ISSUE THREAD

**Issue by puzpuzpuz** [puzpuzpuz/xsync#5]
See https://github.com/golang/go/issues/47643

**Comment by puzpuzpuz** [comment]
Closed with #11

**Comment by costela** [comment]
The current code still mentions the "unshrinkability" of the map:
```
// Also note that, unlike in sync.Map, the underlying hash table used
// by MapOf never shrinks and only grows on demand. However, this
// should not be an issue in many cases since updates happen in-place
// leaving no tombstone entries.
```

AFAIU #11 removed this limitation, correct?

**Comment by puzpuzpuz** [comment]
> The current code still mentions the "unshrinkability" of the map

Yes, that's a left over. Will fix it in the evening.

**Comment by puzpuzpuz** [comment]
Addressed in #37
